### PR TITLE
Fix `is_interactive_compatible` logic after AcceleratorConnector rewrite

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Follow up to #11448 
The PR introduced a bug with Jupyter notebooks because the strategy name property was changed. The check how it is implemented now doesn't make much sense. 

#11970 proposes to move it to the strategy base class itself but my proposal here is to move it to the newly added launcher. IMO this fixes the issue more efficiently. 

[Notebook for manual checking.](https://colab.research.google.com/drive/1DMclmZopMeThbJYcLRYAuzzZqYwq4O4V?usp=sharing)

### Does your PR introduce any breaking changes? If yes, please list them.

No.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @tchaton @rohitgr7 @akihironitta @justusschock @kaushikb11 @awaelchli @ninginthecloud

### CODE REVIEW

File: pytorch_lightning/strategies/launchers/spawn.py
Comment (awaelchli) [inline pytorch_lightning/strategies/launchers/spawn.py:53]: can do this once #11966 refactor is done. 

Here, it should be `return self.start_method == "spawn" or strategy.root_device == cuda`
See #7550 for more context.

File: tests/accelerators/test_accelerator_connector.py
Comment (awaelchli) [inline tests/accelerators/test_accelerator_connector.py:415]: using monkeypatch now because previously `_IS_INTERACTIVE` became a MagicMock, which was fine because it would still take the branch `if _IS_INTERACTIVE:` but not technically correct way of mocking.

Comment (four4fish) [reply tests/accelerators/test_accelerator_connector.py:415]: That's what happend for distermintic! Do we have similar issues in other places?
I remembered TPU_available mock below seems have similar issues.

File: pytorch_lightning/trainer/connectors/accelerator_connector.py
Comment (four4fish) [inline pytorch_lightning/trainer/connectors/accelerator_connector.py:718]: Do you think we should move it to _init_strategy? which fails faster.

Comment (four4fish) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:718]: I was thinking add this to strategy API which could be called at init_strategy. I think in launcher make much more sense. +1

**Review by four4fish — APPROVED** [review]

File: pytorch_lightning/trainer/connectors/accelerator_connector.py
Comment (kaushikb11) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:718]: Why change the variable from `compatible` to `recommended`?

**Review by kaushikb11 — APPROVED** [review]

File: pytorch_lightning/trainer/connectors/accelerator_connector.py
Comment (carmocca) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:718]: +1, let's keep the previous name

File: tests/accelerators/test_accelerator_connector.py
Comment (carmocca) [inline tests/accelerators/test_accelerator_connector.py:418]: What is this testing exactly? Is it that the instantiation works? Can we have some assertions?

**Review by carmocca — APPROVED** [review]
Maybe worth waiting until #11966 is merged so Adrian's comment can be addressed here

File: pytorch_lightning/trainer/connectors/accelerator_connector.py
Comment (awaelchli) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py]: > Do you think we should move it to _init_strategy? which fails faster.

We can't because we need to run `configure_launcher` first and this needs to happen in `lazy_init`.


> Why change the variable from compatible to recommended?

I did this because we decoupled the launchers from the strategy. So the concept of being "compatible with interactive environments" has shifted away from the strategy and it made more sense to recomend (spelling error not intended) strategies that come with compatible launchers by default. Anyway, refactored now.

File: pytorch_lightning/strategies/launchers/spawn.py
Comment (kaushikb11) [reply pytorch_lightning/strategies/launchers/spawn.py:53]: AFAIK, it should work for xla spawn as well

File: tests/accelerators/test_accelerator_connector.py
Comment (kaushikb11) [inline tests/accelerators/test_accelerator_connector.py:413]: ># Edge case: AcceleratorConnector maps dp to ddp if accelerator != gpu

But we should let the users know about it through a warning?

File: pytorch_lightning/utilities/enums.py
Comment (kaushikb11) [inline pytorch_lightning/utilities/enums.py:258]: @awaelchli It's supported, right?

File: pytorch_lightning/strategies/launchers/spawn.py
Comment (awaelchli) [reply pytorch_lightning/strategies/launchers/spawn.py:53]: yeah definitely, for xla spawn we return True.

Comment (awaelchli) [reply pytorch_lightning/strategies/launchers/spawn.py:53]: for the xla spawn launcher, we can just return True

File: tests/accelerators/test_accelerator_connector.py
Comment (awaelchli) [reply tests/accelerators/test_accelerator_connector.py:413]: Well, this is and was the behavior captured by the accelerator connector and associated tests. it is out of scope for this PR for me to add a warning. Here, I'm concerned with raising an exception in an interactive environment.

File: pytorch_lightning/utilities/enums.py
Comment (awaelchli) [reply pytorch_lightning/utilities/enums.py:258]: No, pytorch does no longer support multiprocessing in jupyter notebooks and cuda. only fork+cpu seems to be working according to some tests I did weeks ago. #7550

Comment (awaelchli) [reply pytorch_lightning/utilities/enums.py:258]: added a comment in the code

**Pull request merged**
